### DISCUSSION TITLE
Has stitched mesh syntax changed?  Is it compatible with increase AD container size?

### DISCUSSION THREAD

**Discussion by EdSheltonUKAEA** [idaholab/moose#24744] [Q&A Meshing] [answered]
I'm sure that a few weeks ago I was running models OK with the stitched mesh given below:

> [Mesh]
>   [01] # Regular mesh at surface
>     type = GeneratedMeshGenerator  
>     dim = 1
>     xmin = 0 # m
>     xmax = 5e-8 # m
>     bias_x = 1 
>     nx = 100 # Element size = 5e-10 m
>   []
>   [02] # Biased mesh (for implantation and diffusion)
>     type = GeneratedMeshGenerator  
>     dim = 1
>     xmin = 5e-8 # m
>     xmax = 1e-5 # m
>     bias_x = 1.111676  
>     nx = 73 # First element 4.9e-10 m, last element = 1e-6 m
>   []
>   [03] # Regular mesh to 0.2 mm
>     type = GeneratedMeshGenerator  
>     dim = 1
>     xmin = 1e-5 # m
>     xmax = 2e-4 # m
>     bias_x = 1 # m
>     nx = 190 # Element size = 1e-6 m
>   []
> 
>   [Mesh_stitcher]  
>      type = StitchedMeshGenerator
>      inputs = '01 02 03'
>      stitch_boundaries_pairs = 'right left; right left'
>   []
> []

I'm trying to run it again now, but get error messages:

> In UnstructuredMesh::stitch_meshes:
> This mesh has 1 nodes on boundary 1.
> Other mesh has 2 nodes on boundary 2.
> Minimum edge length on both surfaces is 5e-10.
> In UnstructuredMesh::stitch_meshes:
> Found 1 matching nodes.
> 
> *** ERROR ***
> /home/eshelton/projects/red-kite/problems/HD_3tr_oc616_Ph1Imp_Model.i:38: (Mesh/Mesh_stitcher/stitch_boundaries_pairs):
>     Boundary right doesn't exist on the first mesh in Mesh_stitcher
>     Boundary names that do exist:
>      ID : Name
>      0 : left
>      2 : left

I ran a MOOSE update yesterday.  Has something changed with the StitchedMeshGenerator syntax in the last few weeks which means my old code doesn't work any more.

The other thing that might be different is that in the update I needed to increase AD container size to resolve another error message cause by an increase in number of variables.  I ran this line of code before compiling MOOSE.
> ./configure --with-derivative-size=600 --with-ad-indexing-type=global

Could this have upset the mesh stitcher?

**Comment by GiudGiud** [comment, marked as answer]
Hello

Yes it has changed. 
It used to merge sidesets with the same ids. It does not do it anymore.

If you want the previous behavior, please pass as an additional parameter
prevent_boundary_ids_overlap = false

The AD derivative size is a separate topic

Guillaume

**Reply by EdSheltonUKAEA** [reply]
That's worked great thanks!

But has this also changed something with how the mesh is saved in the an ExodusII ouput file?

When I try to apply the mesh and variable values from the final time step in a subsequent phase of analysis, I get this error:

> Time Step 0, time = 0
> libMesh terminating:
> Error creating ExodusII/Nemesis mesh file.
> [0] ../src/mesh/exodusII_io_helper.C, line 2299, compiled May 15 2023 at 22:49:10
> 
> application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0

Is there a workaround for this?  Or is it some other problem?

This is how I import the mesh from the previous model phases.  It used to work before I did the recent MOOSE update.

> [Mesh]
>   type = FileMesh
>   file = Ph1Imp_out.e
> []

**Reply by GiudGiud** [reply]
Hello

No that part has not changed. 
It seems the error is coming from the "creation" of a new output exodus file.
Can you try silencing the Outputs to exodus for now?

Guillaume

**Reply by EdSheltonUKAEA** [reply]
Ah - looking at that has fixed it.  Because of some updates to my code I was trying to output to an Exodus file with the same name as I was importing from.  Now I've fixed that - it's working!

Completely my error!

Many thanks for this and the fix to original query.
